docs/useRegisterMutationMiddleware.md

```diff
@@ -0,0 +1,100 @@
+---
+layout: default
+title: "useRegisterMutationMiddleware"
+---
+
+# `useRegisterMutationMiddleware`
+
+This hook allows to register a mutation middleware function.
+
+Middleware functions allow you to hook into a mutation process to perform various actions before or after the main mutation action (create or update). They allow similar use cases as the [`withLifeCycleCallbacks`](./withLifecycleCallbacks.md) except they are declared at the component level.
```

Comment:
You should also explain why one may need it: to transform data before saving it, to update / delete related data, etc